crates/astria-conductor/src/celestia/convert.rs

```diff
@@ -64,42 +72,73 @@ pub(super) fn decode_raw_blobs(
 /// An unsorted [`CelestiaSequencerBlob`] and [`CelestiaRollupBlob`].
 pub(super) struct ConvertedBlobs {
     celestia_height: u64,
-    header_blobs: Vec<CelestiaSequencerBlob>,
-    rollup_blobs: Vec<CelestiaRollupBlob>,
+    headers: Vec<CelestiaHeader>,
+    rollup_data_entries: Vec<CelestiaRollupData>,
 }
 
 impl ConvertedBlobs {
-    pub(super) fn len_header_blobs(&self) -> usize {
-        self.header_blobs.len()
+    pub(super) fn len_headers(&self) -> usize {
+        self.headers.len()
     }
 
-    pub(super) fn len_rollup_blobs(&self) -> usize {
-        self.rollup_blobs.len()
+    pub(super) fn len_rollup_data_entries(&self) -> usize {
+        self.rollup_data_entries.len()
     }
 
-    pub(super) fn into_parts(self) -> (u64, Vec<CelestiaSequencerBlob>, Vec<CelestiaRollupBlob>) {
-        (self.celestia_height, self.header_blobs, self.rollup_blobs)
+    pub(super) fn into_parts(self) -> (u64, Vec<CelestiaHeader>, Vec<CelestiaRollupData>) {
+        (self.celestia_height, self.headers, self.rollup_data_entries)
     }
 
     fn new(celestia_height: u64) -> Self {
         Self {
             celestia_height,
-            header_blobs: Vec::new(),
-            rollup_blobs: Vec::new(),
+            headers: Vec::new(),
+            rollup_data_entries: Vec::new(),
         }
     }
 
-    fn push_header(&mut self, header: CelestiaSequencerBlob) {
-        self.header_blobs.push(header);
+    fn push_header(&mut self, header: CelestiaHeader) {
+        self.headers.push(header);
+    }
+
+    fn push_rollup_data(&mut self, rollup: CelestiaRollupData) {
+        self.rollup_data_entries.push(rollup);
     }
 
-    fn push_rollup(&mut self, rollup: CelestiaRollupBlob) {
-        self.rollup_blobs.push(rollup);
+    fn extend_from_header_list_if_welformed(&mut self, list: CelestiaHeaderList) {
+        let initial_len = self.headers.len();
+        if let Err(err) = list.headers.into_iter().try_for_each(|raw| {
+            let header = CelestiaHeader::try_from_raw(raw)?;
+            self.push_header(header);
+            Ok::<(), CelestiaSequencerBlobError>(())
+        }) {
+            info!(
```

Comment:
We are actually doing that! `FetchConvertVerifyAndReconstruct::execute` has the `celestia_height` in its span, and we are then decoding the blobs using this:
```rust
        let decoded_blobs = spawn_blocking(move || {
            decode_span
                .in_scope(|| decode_raw_blobs(new_blobs, rollup_namespace, sequencer_namespace))
        })
        .await
```
Hence all events are associated with the correct celestia height.

`decode_raw_blobs` itself has no knowledge of which Celestia height it is operating under - and both `decode_raw_blobs` and the `extend_from_*_if_well_formed` methods are deliberately not returning errors because we expect that blobs are being posted under our namespaces (no way to prevent that).